pkg/edgecache/finalizer/controller.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package finalizer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/component-base/metrics/prometheus/ratelimiter"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/blob-csi-driver/pkg/edgecache"
+	"sigs.k8s.io/blob-csi-driver/pkg/util"
+)
+
+const (
+	finalizerName          string = "external/blob-csi-driver-edgecache"
+	pvcProtectionFinalizer string = "kubernetes.io/pvc-protection"
+	accountAnnotation      string = "external/edge-cache-account"
+	containerAnnotation    string = "external/edge-cache-container"
+)
+
+// Controller is controller that removes PVProtectionFinalizer
+// from PVs that are not bound to PVCs.
+type Controller struct {
+	client clientset.Interface
+
+	pvcLister       corelisters.PersistentVolumeClaimLister
+	pvcListerSynced cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+
+	eventRecorder record.EventRecorder
+
+	ecManager edgecache.ManagerInterface
+}
+
+// NewHydraFinalizerController returns a new *Controller.
+func NewHydraFinalizerController(manager edgecache.ManagerInterface, pvcInformer coreinformers.PersistentVolumeClaimInformer, client clientset.Interface) *Controller {
```

Comment:
please update per naming conventions

Reply:
updated to use 'edgecache' consistently and try to improve log verbosity and content

Reply:
NewHydraFinalizerController is still a function name.  I think that should be NewEdgeCacheFinalizerController?